src/fluree/store/core.cljc

```diff
@@ -0,0 +1,69 @@
+(ns fluree.store.core
+  (:refer-clojure :exclude [read])
+  (:require [fluree.store.proto :as store-proto]
+            [fluree.store.file :as file-store]
+            [fluree.store.localstorage :as localstorage-store]
+            [fluree.store.memory :as mem-store]
+            [malli.core :as m]))
+
+(def BaseConfig
+  [:map
+   [:store/method [:enum :memory :localstorage :file :ipfs :s3 :remote]]])
+
+(def FileConfig
+  [:and
+   BaseConfig
+   [:map
+    [:store/method [:enum :file]]
+    [:file-store/storage-path :string]]])
+
+(def LocalStorageConfig
+  [:and
+   BaseConfig
+   [:map
+    [:store/method [:enum :localstorage]]]])
+
+(def MemoryConfig
+  [:and
+   BaseConfig
+   [:map
+    [:store/method [:enum :memory]]
+    [:memory-store/storage-atom {:optional true} :any]]])
+
+(def StoreConfig
+  [:or
+   FileConfig
+   LocalStorageConfig
+   MemoryConfig])
+
+(defn start
+  [{:keys [:store/method] :as config}]
+  (if-let [config-error (m/explain StoreConfig config)]
+    (throw (ex-info "Invalid Store config."
+                    {:error  config-error
+                     :config config}))
+    (case method
```

Comment:
maybe not for this pr, we should use an open multi method instead of a closed case statement to decide which storage implementation to load. this allows for more flexibility in which implementations we support, even user supplied ones.